### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by Mukumbuta**
### Project requirements
    - Created a class `Classroom` , `Students`, `Rental` and `Book` with necessary instance variables iinitialized in the constructor.
    - Setter and getter for @label  using `attr_accessor`
    - Created the has-many/belongs-to relationship between Classroom and Student. The following features are implemented:
          - Make sure that when adding a student to a classroom it also sets the classroom for the student 
          - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
   - Create a class  with the following:
   - Create the `many-to-many` relationship between `Person` and `Book` using the intermediate class `Rental`.            
   - Create the has-many side of `Book` and `Rental` (a book has many rentals) 👍🏾 
   - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book) 👍🏾 
   - Create the `has-many` side of `Person` and `Rental` (a person has many rentals) 👍🏾 
   - Create the `belongs-to `side of `Rental` and `Person` (a rental belongs to a person) 👍🏾 
   - Modify the constructor of Rental so `Book` and `Person` are set in it 👍🏾 
   - fix all linter errors

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Make sure that when adding a student to a classroom it also sets the classroom for the student
+# Make sure that when setting the classroom for a student it also adds it to the classrooms students
+class Classroom
+  attr_accessor :label, :student
```
Comment (rloterh) [inline classroom.rb]: - [ ] Following best practices, we only intend to _access (get)_ `students` object and therefore it should be marked with `attr_reader`. Marking it with `attr_accessor` may expose our object to unwanted modifications. Kindly ensure to improve upon this aspect.

File: students.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Named Students to differentiate it from the already existing Student class
+class Students
+  attr_accessor :name, :classroom, :rentals
```
Comment (rloterh) [inline students.rb]: - [ ] It's a good practice we mark the _classroom_ object with `attr_reader`, as we only intend to _access (get)_. this object. Doing so will prevent the _classroom_ object from unnecessary modifications.

File: students.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
```
Comment (rloterh) [inline students.rb:6]: - [ ] _Student_ object belongs to the _Classroom_. Let's endeavor to establish a belong to association accordingly like:
```ruby
class Student < Person
  attr_reader :classroom
  .
  .
   def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
end
```

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Book class
+class Book
+  attr_accessor :title, :author, :rentals
+
```
Comment (rloterh) [inline book.rb:4]: - [ ] A _rental_ can have many _books_, and _books_ can be subjected to many _rentals_. Let's therefore establish a _many-to-many_ association between _rentals_ and _books_ (let's bear in mind that the association between _person_ and _rentals_ is also _many-to-many_) by defining a method for adding rentals within the book object like below:

```ruby
class Book
  attr_accessor :title, :author, :rentals
  .
  .

  def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: students.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Named Students to differentiate it from the already existing Student class
+class Students
```
Comment (rloterh) [inline students.rb]: - [ ] [OPTIONAL] 
A _student_ object inherits from the _person_ class. Let's endeavor to declare this relationship.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Mukumbuta, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🚦  🚦  🏃 

## Highlights 👍  
- Good job so far establishing associations for a book library objects! 

## Required Changes ♻️
- [ ] A _person_ has many _rentals_, and this association is missing in your _person_ class. Let's define this association by ensuring:
  * Getters and Setters for @rentals is present
  * Initialize @rentals
  * Defining a method for adding rentals
  * The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

This should be similar to the below code example:
```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def initialize(age, name = "Unknown", parent_permission = true)
    .
    .
   
    @rentals = []
  end

    Rental.new(date, book, self)
  end

  .
  .
end
```
You may read further regarding the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_principles_examples.md#useful-shortcuts-attr_reader-attr_writer-and-attr_accessor).

- [ ] MIT License link your ReadMe is broken, kindly endeavor to fix it!

_Kindly check the other inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: students.rb (hunk 3 above)
Comment (rotimiazeez) [reply students.rb:6]: Kindly establish the belong to association as suggested by the CR. 😉

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
Hi @Mukumbuta  😉, 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Suggested changes
### The MIT License link provided in your README.MD file is broken. Kindly fix this as the previous CR suggested 👌🏽
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me(@rotimiazeez) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hi there 👋🏽 , 

Great work on making the changes requested by a previous reviewer 👏 
To highlight 👍🏽 
✔️ Associations have been created

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Required Changes ♻️
- [ ] Like the previous reviewers mentioned, you are still missing the belongs-to relationship between classroom and student. You can read more about it [here](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships). It should look something like this:

```ruby
class Student < Person
  attr_reader :classroom
  .
  .
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
end
```
- Also, you currently have two student classes, I suggest you remove one of them to avoid confusion as only one is needed.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @Mukumbuta, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**